subsys/net/lib/nrf_provisioning/src/nrf_provisioning.c

```diff
@@ -251,6 +244,65 @@ static int nrf_provisioning_set(const char *key, size_t len_rd,
 	return -ENOENT;
 }
 
+int nrf_provisioning_notify_event_and_wait_for_modem_state(
+				int timeout_seconds,
+				enum nrf_provisioning_event event,
+				nrf_provisioning_event_cb_t callback)
+{
+	int ret;
+	enum lte_lc_func_mode fmode;
+
+	if (callback == NULL) {
+		LOG_ERR("Callback data is NULL");
+		return -EINVAL;
+	}
+
+	if ((event != NRF_PROVISIONING_EVENT_NEED_LTE_DEACTIVATED) &&
+	    (event != NRF_PROVISIONING_EVENT_NEED_LTE_ACTIVATED)) {
+		LOG_ERR("Invalid event");
+		return -EINVAL;
+	}
+
+	struct nrf_provisioning_callback_data event_data = {
+		.type = event,
+	};
+
+	callback(&event_data);
+
+	for (int i = 0; i < timeout_seconds; i++) {
+		ret = lte_lc_func_mode_get(&fmode);
+		if (ret) {
+			LOG_ERR("Failed to read modem functional mode");
+			return ret;
+		}
+
+		if (event == NRF_PROVISIONING_EVENT_NEED_LTE_DEACTIVATED) {
+			if (fmode == LTE_LC_FUNC_MODE_OFFLINE) {
+				LOG_DBG("Modem is offline");
+				return 0;
+			}
```

Comment:
Connection manager disconnects only LTE and GNSS may still be active. Please check also if the functional mode is LTE_LC_FUNC_MODE_ACTIVATE_GNSS. Then deactivate the GNSS or send new event to the application which can deactivate the GNSS.
Related PR https://github.com/nrfconnect/sdk-nrf/pull/24177

Reply:
I could update the description for

```c
/**
 * Credential handling requires LTE to be deactivated.
 * Application should call lte_lc_offline() or similar to deactivate LTE
 * connectivity. This event is generated when the library needs to write
 * certificates or credentials to the modem, which requires the LTE connection
 * to be offline.
 */
NRF_PROVISIONING_EVENT_NEED_LTE_DEACTIVATED,
```
	
to state to warn about GNSS being active.

Reply:
Done

Reply:
> Alright, will do. Does the sample even enable gnss? Or is the check meant as a safeguard in case it has been enabled through at shell or something?

The sample doesn't initially enable GNSS. However, toggling the modem to flight mode and back to normal mode (using shell or AT commands) activates GNSS, as GNSS support is enabled by default. As a result, provisioning credentials fail to timeout because, after the connection manager disconnects LTE, the modem remains in LTE_LC_FUNC_MODE_ACTIVATE_GNSS functional mode.

Reply:
Ok, thanks. Ive added the check and docs.